src/box/memtx_tree.cc

```diff
@@ -407,11 +407,71 @@ tree_iterator_prev_equal_base(struct iterator *iterator, struct tuple **ret)
 	return 0;
 }
 
+const char *
+to_mp_array(struct txn *txn, const char *key, uint32_t part_count)
+{
+	if (!txn || !key)
+		return key;
+
+	const char *end = key;
+	for (size_t i = 0; i < part_count; ++i) {
+		mp_next(&end);
+	}
+
+	size_t size = end - key + mp_sizeof_array(part_count);
+	char *new_key = (char *) region_alloc(&txn->region, size);
```

Comment:
It is necessary to check the pointer for NULL, otherwise in case of insufficient memory tarantool crashed